src/utils.ts

```diff
@@ -0,0 +1,137 @@
+import { existsSync, mkdirSync, readFileSync, writeFileSync } from "node:fs";
+import { dirname, join } from "node:path";
+import { fileURLToPath } from "node:url";
+import { parseArgs } from "node:util";
+import fetch from "node-fetch";
+import { ImportMap } from "@jspm/import-map";
+import { CreateCacheMapFactory } from "src/types";
+import {
+  ALL_CACHE_MAP_REQUIREMENTS_MUST_BE_DEFINED,
+  NO_CACHE_MAP_DEFINED,
+  PROCESS_CLI_ARGS_OPTIONS,
+} from "src/constants";
+
+/**
+ * ******************************************************
+ * UTILS
+ * ------------------------------------------------------
+ * @description a collection of utility functions
+ * @summary a collection of utility functions
+ * which aren't reliant on each other (for easy testability)
+ *
+ * ******************************************************
+ */
+
+const cwd = process.cwd();
+
+/**
+ * constructPath
+ * @description a convenience function to construct a path
+ * @param {string} dir
+ * @param {string} root
+ * @returns {string}
+ */
+export const constructPath = (dir: string, root = "./") => join(root, dir);
+
+/**
+ * constructUrlPath
+ * @description a convenience function to construct a url path
+ * @param {string} base
+ * @param {string} url
+ * @param {boolean} debug
+ * @returns {string}
+ */
+export const constructUrlPath = (base = ".", url: string = cwd, debug = false) => {
+  try {
+    const path = new URL(base, url);
+    return fileURLToPath(path.href);
+  } catch (err) {
+    if (debug) console.error(`constructUrlPath: Failed in creating a url path ${err}`);
+    return "";
+  }
+};
+
+/**
+ * constructImportMap
+ * @description a convenience function to construct an import map
+ * @param {string} path
+ * @param {string} rootUrl
+ * @returns {ImportMap|null}
+ */
+export const constructImportMap = (path = "", rootUrl = cwd) => {
+  const pathExists = existsSync(path);
+  const json = readFileSync(path, { encoding: "utf8" });
+  const map = pathExists && json ? JSON.parse(json) : {};
+  return new ImportMap({
+    rootUrl,
+    map,
+  });
+};
+
+/**
+ * createCacheMap
+ * @description a factory function for managing a cache map
+ * @param {boolean} debug
+ * @returns {createCacheMap}
+ */
+export const createCacheMap = (debug = false): CreateCacheMapFactory => {
+  const instance = new Map();
+  const isDebugging = debug;
+  let cachePath: string | undefined, modulePath: string | undefined;
+  return {
+    cachePath,
+    instance,
+    isDebugging,
+    modulePath,
+    get(cachePath: string) {
+      const path = this.instance.get(cachePath);
+      if (!path) {
+        if (this.isDebugging) console.error(NO_CACHE_MAP_DEFINED);
```

Comment:
Just a small change, you can extract this out as a `loggger` and then just pass the content to it all the time. 👍